From 7cb59e009a6ac55a05ee115737363af2349dbf71 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sat, 12 Sep 2020 20:30:49 +0200
Subject: [PATCH 2/2] Enable pkg-config for mingw-w64 build

---
 libmariadb/CMakeLists.txt | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
index d6e23e56..67a93e31 100644
--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -501,19 +501,14 @@ ENDIF()
 INSTALL(TARGETS mariadbclient
           COMPONENT Development
           DESTINATION ${INSTALL_LIBDIR})
-IF(WIN32)
-INSTALL(TARGETS libmariadb
+IF(WIN32 AND NOT MINGW)
+INSTALL(TARGETS mariadb
         COMPONENT SharedLibraries
         DESTINATION ${INSTALL_LIBDIR})
 ELSE()
 # in cmake 3.12+ we can use
-#INSTALL(TARGETS mariadb LIBRARY DESTINATION ${INSTALL_LIBDIR} RUNTIME DESTINATION ${INSTALL_BINDIR} ARCHIVE DESTINATION ${INSTALL_LIBDIR}
-#        COMPONENT SharedLibraries NAMELINK_COMPONENT Development)
-# but as long as we build on CentOS 7 with its cmake 2.8.12.2 we have to use
-INSTALL(TARGETS mariadb LIBRARY DESTINATION ${INSTALL_LIBDIR} RUNTIME DESTINATION ${INSTALL_BINDIR} ARCHIVE DESTINATION ${INSTALL_LIBDIR}
-        COMPONENT SharedLibraries NAMELINK_SKIP)
-INSTALL(TARGETS mariadb LIBRARY DESTINATION ${INSTALL_LIBDIR} RUNTIME DESTINATION ${INSTALL_BINDIR} ARCHIVE DESTINATION ${INSTALL_LIBDIR}
-        COMPONENT Development NAMELINK_ONLY)
+INSTALL(TARGETS mariadb RUNTIME DESTINATION ${INSTALL_BINDIR} ARCHIVE DESTINATION ${INSTALL_LIBDIR} LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        COMPONENT SharedLibraries NAMELINK_COMPONENT Development)
 ENDIF()
 
 IF(MSVC)
-- 
2.46.0

