From e0502aec7784f655db71380f375e959debacaa35 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 19 Mar 2017 18:04:48 +0100
Subject: [PATCH 2/2] Adjust service file for Arch Linux package

---
 scripts/systemd/gitea.service | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/scripts/systemd/gitea.service b/scripts/systemd/gitea.service
index 758f1590..302e7a9e 100644
--- a/scripts/systemd/gitea.service
+++ b/scripts/systemd/gitea.service
@@ -15,12 +15,12 @@ After=network.target
 #LimitMEMLOCK=infinity
 #LimitNOFILE=65535
 Type=simple
-User=git
-Group=git
-WorkingDirectory=/home/git/gitea
-ExecStart=/home/git/gitea/gitea web
+User=gitea
+Group=gitea
+WorkingDirectory=/var/lib/gitea
+ExecStart=/usr/bin/gitea web --config /etc/gitea/app.ini
 Restart=always
-Environment=USER=git HOME=/home/git
+Environment=USER=gitea HOME=/var/lib/gitea
 
 [Install]
 WantedBy=multi-user.target
-- 
2.13.1

