diff -Naur qtmultimedia-opensource-src-5.7.0.orig/src/plugins/common/evr/evrd3dpresentengine.cpp qtmultimedia-opensource-src-5.7.0/src/plugins/common/evr/evrd3dpresentengine.cpp
--- qtmultimedia-opensource-src-5.7.0.orig/src/plugins/common/evr/evrd3dpresentengine.cpp	2016-06-08 06:38:56.000000000 -0500
+++ qtmultimedia-opensource-src-5.7.0/src/plugins/common/evr/evrd3dpresentengine.cpp	2016-11-03 14:07:13.172478447 -0500
@@ -49,9 +49,9 @@
 #include <private/qmediaopenglhelper_p.h>
 
 #ifdef MAYBE_ANGLE
-# include <qtgui/qguiapplication.h>
-# include <qpa/qplatformnativeinterface.h>
-# include <qopenglfunctions.h>
+# include <QtGui/qguiapplication.h>
+# include <QtGui/qpa/qplatformnativeinterface.h>
+# include <QtGui/qopenglfunctions.h>
 # include <EGL/eglext.h>
 #endif
 
