--- model/CMakeLists.txt.sav	2021-10-06 09:02:03.463489669 +0200
+++ model/CMakeLists.txt	2021-10-06 09:02:11.203489460 +0200
@@ -48,7 +48,7 @@
 use_syncthingconnector(VISIBILITY PUBLIC)
 
 # find qtforkawesome
-find_package(${PACKAGE_NAMESPACE_PREFIX}qtforkawesome${CONFIGURATION_PACKAGE_SUFFIX_QTFORKAWESOME} 0.0.1 REQUIRED)
+find_package(${PACKAGE_NAMESPACE_PREFIX}qtforkawesome-git${CONFIGURATION_PACKAGE_SUFFIX_QTFORKAWESOME} 0.0.1 REQUIRED)
 use_qt_fork_awesome(VISIBILITY PUBLIC)
 
 # link also explicitly against the following Qt modules
