From f3db8f65bb0f3fd538dad81667354b963900072c Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 9 Dec 2018 14:31:30 +0100
Subject: [PATCH 2/2] Fix androiddeployqt search paths

---
 src/tools/androiddeployqt/main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tools/androiddeployqt/main.cpp b/src/tools/androiddeployqt/main.cpp
index 76c521ca8f..0b0ba8dcae 100644
--- a/src/tools/androiddeployqt/main.cpp
+++ b/src/tools/androiddeployqt/main.cpp
@@ -1686,7 +1686,7 @@ bool scanImports(Options *options, QSet<QString> *usedDependencies)
         rootPath += QLatin1Char('/');
 
     QStringList importPaths;
-    importPaths += shellQuote(options->qtInstallDirectory + QLatin1String("/qml"));
+    importPaths += shellQuote(options->qtInstallDirectory + QLatin1String("/lib/qt/qml"));
     importPaths += rootPath;
     for (const QString &qmlImportPath : qAsConst(options->qmlImportPaths))
         importPaths += shellQuote(qmlImportPath);
@@ -1817,7 +1817,7 @@ bool readDependencies(Options *options)
         return false;
 
     // Jam in the dependencies of the platform plugin, since the application will crash without it
-    if (!readDependenciesFromElf(options, options->qtInstallDirectory + QLatin1String("/plugins/platforms/android/libqtforandroid.so"), &usedDependencies, &remainingDependencies))
+    if (!readDependenciesFromElf(options, options->qtInstallDirectory + QLatin1String("/lib/qt/plugins/platforms/android/libqtforandroid.so"), &usedDependencies, &remainingDependencies))
         return false;
 
     while (!remainingDependencies.isEmpty()) {
-- 
2.20.1

