From fc643d716b33387759d9be0eeff250deb624d5e9 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 7 Apr 2019 09:54:49 +0200
Subject: [PATCH 3/3] Support pkg-config under Android

from https://aur.archlinux.org/cgit/aur.git/commit/?h=android-aarch64-qt5&id=d901da324b2ceea2ac99b48137cb82b2bd920c77
---
 mkspecs/android-clang/qmake.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mkspecs/android-clang/qmake.conf b/mkspecs/android-clang/qmake.conf
index 5cc6ee79e2..c731b4ef23 100644
--- a/mkspecs/android-clang/qmake.conf
+++ b/mkspecs/android-clang/qmake.conf
@@ -57,3 +57,6 @@ else: \
 QMAKE_CFLAGS_OPTIMIZE_SIZE = -Oz
 
 include(../common/android-base-tail.conf)
+
+QMAKE_PKG_CONFIG = $$[QT_INSTALL_PREFIX]
+QMAKE_PKG_CONFIG = android-$$section(QMAKE_PKG_CONFIG, /, 3, 3)-pkg-config
-- 
2.21.0

